dnf5/include/dnf5/shared_options.hpp

```diff
@@ -58,6 +58,29 @@ class SkipUnavailableOption : public libdnf::cli::session::BoolOption {
               &command.get_context().base.get_config().get_skip_unavailable_option()) {}
 };
 
+class AllowDowngradeOption : public libdnf::cli::session::BoolOption {
+public:
+    explicit AllowDowngradeOption(dnf5::Command & command)
+        : BoolOption(
+              command,
+              "allow-downgrade",
+              '\0',
+              _("Allow downgrade of dependencies to resolve requested transaction"),
+              false,
+              &command.get_context().base.get_config().get_allow_downgrade_option()) {}
+};
+
+class NoDowngradeOption : public libdnf::cli::session::BoolOption {
+public:
+    explicit NoDowngradeOption(dnf5::Command & command)
+        : BoolOption(
+              command,
+              "no-downgrade",
+              '\0',
+              _("Disable downgrade of dependencies"),
+              true,
+              &command.get_context().base.get_config().get_allow_downgrade_option()) {}
+};
```

Comment:
FWIW, I'd prefer having a pair of options, one of them prefixed by `no-`. E.g. `allow-downgrade / no-allow-downgrade`

Reply:
And they should be mutually exclusive.

Reply:
`no-alloew-downgrade` is used by zipper. I also would like to have them exclusive, but this is still under development